components/channel_layout/channel_identifier_router/actions.js

```diff
@@ -86,14 +86,19 @@ function goToChannelByChannelId(match, history) {
     };
 }
 
-function goToChannelByChannelName(match, history) {
+export function goToChannelByChannelName(match, history) {
     return async (dispatch, getState) => {
         const state = getState();
         const {team, identifier} = match.params;
         const channelName = identifier.toLowerCase();
 
-        let channel = getChannelByName(state, channelName);
         const teamObj = getTeamByName(state, team);
+        if (!teamObj) {
```

Comment:
Just curious why the check for team here? This would be the case for manually changing the url to a nonexistent teamName?

Reply:
Yeah that would be the case or if for whatever reason the team wasn't pulled locally (which should never really be the case as we pull all your teams locally on load). Just didn't want to access an object without checking it's existence first

Reply:
In any real path to this code it's already handled up here https://github.com/mattermost/mattermost-webapp/blob/f3bb1501d361554d352f3a6103840135da8b7737/components/channel_layout/channel_identifier_router/actions.js#L24 as well